src/Jobs/GenerateTermSitemapJob.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Rapidez\Statamic\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Rapidez\Statamic\Actions\GenerateSitemapAction;
+use Statamic\Facades\Taxonomy as TaxonomyFacade;
+use Statamic\Sites\Site;
+use Statamic\Taxonomies\LocalizedTerm;
+use Statamic\Taxonomies\Taxonomy;
+
+class GenerateTermSitemapJob implements ShouldQueue, ShouldBeUnique
+{
+    use Dispatchable, InteractsWithQueue, Queueable;
+
+    protected GenerateSitemapAction $sitemapGenerator;
+
+    public function __construct(
+        protected Site $site,
+        protected Taxonomy $taxonomy,
+    ) {
+        $this->sitemapGenerator = new (GenerateSitemapAction::class);
+    }
+
+    public function handle(): void
+    {
+        $sitemap = $this->sitemapGenerator->createSitemap($this->generateTermSitemap());
+
+        file_put_contents(public_path('sitemap_statamic_taxonomy_' . $this->site->handle() . '_' . $this->taxonomy->handle() . '.xml'), $sitemap);
```

Comment:
It's safer to use the [Laravel Storage facade](https://laravel.com/docs/11.x/filesystem#storing-files) for this